src/components/SellNft/index.tsx

```diff
@@ -1,28 +1,137 @@
-import React from 'react';
+import React, { useState }  from 'react'
 import { useForm, Controller } from 'react-hook-form';
-import { LAMPORTS_PER_SOL } from '@solana/web3.js';
 import { Link } from 'react-router-dom';
+import { useConnection, useWallet } from '@solana/wallet-adapter-react'
+import { NATIVE_MINT } from '@solana/spl-token'
+import { AuctionHouseProgram } from '@metaplex-foundation/mpl-auction-house'
+import { Transaction, PublicKey, LAMPORTS_PER_SOL } from '@solana/web3.js'
+import BN from 'bn.js'
+
+const { createSellInstruction } = AuctionHouseProgram.instructions
 
 interface OfferForm {
   amount: string;
 }
 
 interface OfferProps {
   nft: any;
+  ah: any;
 }
 
 const SellNft = ({ nft }: OfferProps) => {
   const { control, watch } = useForm<OfferForm>({});
+  const { publicKey, signTransaction } = useWallet()
+  const { connection } = useConnection()
+  const [sellAmount, setSellAmount] = useState(0)
+
+  const sellNftTransaction = async () => {
+    const sellPrice = String(Number(sellAmount) * LAMPORTS_PER_SOL)
+    const tokenSize = '1'
+    const auctionHouse = new PublicKey(nft.auctionHouse.address)
+    const authority = new PublicKey(nft.auctionHouse.authority)
+    const auctionHouseFeeAccount = new PublicKey(nft.auctionHouse.auctionHouseFeeAccount)
+
+    const tokenMint = new PublicKey(nft.mintAddress)
+
+
+    if (!publicKey || !signTransaction) {
+      return
+    }
+
+    const associatedTokenAccount = (
+      await AuctionHouseProgram.getAtaForMint(tokenMint, new PublicKey(nft.owner.address)) 
+    )[0] 
+
+
+    // Find TradeState Account
+    const [
+      sellerTradeState,
+      tradeStateBump,
+    ] = await AuctionHouseProgram.findTradeStateAccount(
+      publicKey,
+      auctionHouse,
+      associatedTokenAccount, // VERIFY THIS
+      NATIVE_MINT,
+      tokenMint,
+      sellPrice,
+      tokenSize
+    )
+
+
+    const metadata = await AuctionHouseProgram.getMetadata(tokenMint)
```

Comment:
This has been removed. Can get this helper by including the mp-token-metadata package.